app/controllers/books_controller.rb

```diff
@@ -1,5 +1,6 @@
 class BooksController < ApplicationController
   before_action :set_book, only: %i[show edit update destroy]
+  before_action :authenticate_user!
```

Comment:
Since this will be in every controller, I would be better to place it in `ApplicationController`